GitCommands/Git/UserExternalOperationException.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System;
+
+#nullable enable
+
+namespace GitCommands
+{
+    /// <summary>
+    /// Represents errors that occur during execution of user-configured operation, e.g. a script.
+    /// </summary>
+    public class UserExternalOperationException : ExternalOperationException
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="UserExternalOperationException"/> class with a specified parameters
+        /// and a reference to the inner exception that is the cause of this exception.
+        /// </summary>
+        /// <param name="command">The command that led to the exception.</param>
+        /// <param name="arguments">The command arguments.</param>
+        /// <param name="workingDirectory">The working directory.</param>
+        /// <param name="innerException">The exception that is the cause of the current exception.</param>
+        public UserExternalOperationException(string command, string arguments, string workingDirectory, Exception innerException)
```

Comment:
I don't think we can do this without breaking the hierarchy: `public class UserExternalOperationException : ExternalOperationException`.
In order to initialise the exception we need to pass args to the base class, and we only use `context` we won't have information to pass.

Likewise in event where we use this exception, we no longer interested in the original `ExternalOperationException`, but interested in the underlying exception that triggered the sequence:
```cs
catch (ExternalOperationException ex)
{
    throw new UserExternalOperationException(ex.Command, ex.Arguments, ex.WorkingDirectory, ex.InnerException ?? ex);
}
```

Reply:
The /  my goal is to avoid all those redundant lengthy `catch` blocks and to provide information about the top-level operation.

```
public UserExternalOperationException(string context, ExternalOperationException ex)
    : base(ex.Command, ex.Arguments, ex.WorkingDirectory, ex.InnerException)
{
    Context = context;
}

foo()
{
    try
    {
         ...
    }
    catch (ExternatlOperationException ex)
    {
        throw new UserExternalOperationException("context information", ex);
    }
}
```